omod/src/test/java/org/openmrs/module/idgen/web/controller/IdentifierSourceRestControllerTest.java

```diff
@@ -0,0 +1,185 @@
+	@Before
+	public void setUp() throws Exception {
+		executeDataSet("org/openmrs/module/idgen/include/TestDataForLoadTest.xml");
+		this.service = Context.getService(IdentifierSourceService.class);
+	}
+
+	@Override
+	public String getURI() {
+		return "idgen/identifiersource";
+	}
+
+	@Override
+	public String getUuid() {
+		return IDENTIFIER_SOURCE_UUID;
+	}
+
+	@Override
+	public long getAllCount()  {
+		return service.getAllIdentifierSources(false).size();
+	}
+	
+	@Test
+	public void shouldEditAnIdentifierSource() throws Exception {
+		
+		String description = "{\"description\":\"Updated description\"}";
+		
+		handle(newPostRequest(getURI() + "/" + getUuid(), description));
+		
+		assertEquals("Updated description", service.getIdentifierSourceByUuid(getUuid()).getDescription());
+		
+	}
+	
+	@Test
+	public void shouldVoidAnIdentifierSource() throws Exception {
+		
+		final String reason = "none";
+		
+		assertEquals(false, service.getIdentifierSourceByUuid(getUuid()).isRetired());
+		
+		MockHttpServletRequest req = request(RequestMethod.DELETE, getURI() + "/" + getUuid());
+		req.addParameter("reason", reason);
+		handle(req);
+		
+		assertEquals(true, service.getIdentifierSourceByUuid(getUuid()).isRetired());
+		assertEquals(reason, service.getIdentifierSourceByUuid(getUuid()).getRetireReason());
+
+	}
+	
+	@Test
+	public void shouldPurgeAnIdentifierSource() throws Exception {
+		
+		assertNotNull(service.getIdentifierSourceByUuid(getUuid()));
+		
+		handle(newDeleteRequest(getURI() + "/" + getUuid(), new Parameter("purge", "")));
+		
+		assertNull(service.getIdentifierSourceByUuid(getUuid()));
+		
+	}	
+	
+	// error: ERROR! org.openmrs.module.webservices.rest.web.response.ConversionException: 
+	// converting class org.openmrs.module.idgen.SequentialIdentifierGenerator to 
+	// org.openmrs.module.webservices.rest.web.representation.RefRepresentation@ff5aa42
+	
+	@Test
+	public void shouldGetAll() throws Exception  {
+		
+	//	SimpleObject result = deserialize(handle(newGetRequest(getURI())));
+	//		
+	//	assertEquals(getAllCount(), Util.getResultsSize(result));
+		
+	}
+	
+	// error: ERROR! org.codehaus.jackson.map.JsonMappingException: 
+	// Conflicting getter definitions for property "retired": 
+	// org.openmrs.PatientIdentifierType_$$_javassist_23#getRetired(0 params) vs 
+	// org.openmrs.PatientIdentifierType_$$_javassist_23#isRetired(0 params)
+
+	@Test
+	public void shouldGetAnIdentifierSourceByUuid() throws Exception {
+		
+	//	SimpleObject result = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+	//			
+	//	assertNotNull(result);
+	//	assertEquals(getUuid(), PropertyUtils.getProperty(result, "uuid"));
+	
+	}
+		
+	// error: ERROR! org.openmrs.module.webservices.rest.web.response.ConversionException: 
+	// identifierType on class org.openmrs.module.idgen.SequentialIdentifierGenerator
+
+	@Test
+	public void shouldSaveIdentifierSource() throws Exception {
+		
+	//	long initialIdentifierSourceCount = getAllCount();
```

Comment:
Please uncomment all this code. Its ok for the pull request to have test failures at this stage because am reviewing it to guide you on how to proceed.

Reply:
I have uncommented all the code.